src/database/UIDatabase.js

```diff
@@ -6,6 +9,32 @@ export class UIDatabase {
     this.database = database;
   }
 
+  /**
+   * Closes the database, exports the .realm file to android 'Download/mSupplyMobile data'
+   * The app will cease to function (crash) if anything tries to access the database while
+   * it is closed.
+   *
+   * This method should perhaps be added to react-native-databse
+   */
+  exportData(filename = 'msupply-mobile-data') {
+    const realm = this.database.realm; // TODO: refactor away from 'database.database.realm'
+    const realmPath = realm.path;
+    const exportFolder = `${RNFS.ExternalStorageDirectoryPath}/Download/mSupplyMobile_data`;
+
+    const date = new Date();
+    const dateString = `${date.getFullYear()}-${date.getMonth()}-${date.getDay()}T${date.getHours()}-${date.getMinutes()}-${date.getSeconds()}`;
+    const copyFileName = `${filename} ${dateString}`;
+
+    // If the database is not closed, there is a small chance of corrupting the data
+    // if it's currently in a transaction
+    realm.close();
+    RNFS.mkdir(exportFolder)
+      .then(() => RNFS.copyFile(realmPath, `${exportFolder}/${copyFileName}.realm`)
+        .catch(() => RNFS.copyFile(realmPath, `${exportFolder}/msupply-mobile-data.realm`))
```

Comment:
hmm, I don't see why this catch is necessary

Reply:
If I recall correctly it's due to `copyFileName` being derived from store name, it can have invalid characters for a filename/path. Added a comment.